Build against lts-15.5 fails

When building with stack against lts-15.5 (ghc-8.8.3), build fails with:

```
yeshql-core       > /private/var/folders/np/jf9827dj7290rd3dxg244_7r0000gn/T/stack83190/yeshql-core-4.1.0.2/src/Database/YeshQL/Util.hs:89:27: error:
yeshql-core       >     • Could not deduce (MonadFail m) arising from a use of ‘fail’
yeshql-core       >       from the context: (Monad m, Show e)
yeshql-core       >         bound by the type signature for:
yeshql-core       >                    withParsed :: forall (m :: * -> *) e s a b.
yeshql-core       >                                  (Monad m, Show e) =>
yeshql-core       >                                  (s -> Either e a) -> (a -> m b) -> s -> m b
yeshql-core       >         at src/Database/YeshQL/Util.hs:(84,1)-(85,57)
yeshql-core       >       Possible fix:
yeshql-core       >         add (MonadFail m) to the context of
yeshql-core       >           the type signature for:
yeshql-core       >             withParsed :: forall (m :: * -> *) e s a b.
yeshql-core       >                           (Monad m, Show e) =>
yeshql-core       >                           (s -> Either e a) -> (a -> m b) -> s -> m b
yeshql-core       >     • In the first argument of ‘(.)’, namely ‘fail’
yeshql-core       >       In the expression: fail . show
yeshql-core       >       In the expression: fail . show $ e
yeshql-core       >    |
yeshql-core       > 89 |                 Left e -> fail . show $ e
yeshql-core       >    |                           ^^^^
yeshql-core       >    
```

--- a/core/src/Database/YeshQL/Util.hs
+++ b/core/src/Database/YeshQL/Util.hs
@@ -57,15 +57,15 @@ nameQueries basename queries =
     where
         queryNames = [ basename ++ "_" ++ show i | i <- [0..] ]
 
-withParsedQuery :: (MonadPerformIO m, Monad m)
+withParsedQuery :: (MonadFail m, MonadPerformIO m, Monad m)
                 => (ParsedQuery -> m a) -> String -> m a
 withParsedQuery = withParsed parseQuery
 
-withParsedQueries :: (MonadPerformIO m, Monad m)
+withParsedQueries :: (MonadFail m, MonadPerformIO m, Monad m)
                   => ([ParsedQuery] -> m a) -> String -> m a
 withParsedQueries = withParsed parseQueries
 
-withParsedQueryFile :: (MonadPerformIO m, Monad m)
+withParsedQueryFile :: (MonadFail m, MonadPerformIO m, Monad m)
                     => (ParsedQuery -> m a) -> FilePath -> m a
 withParsedQueryFile p fn =
     withParsedFile
@@ -73,7 +73,7 @@ withParsedQueryFile p fn =
         (p . nameQuery (queryIdentifier "" fn))
         fn
 
-withParsedQueriesFile :: (MonadPerformIO m, Monad m)
+withParsedQueriesFile :: (MonadFail m, MonadPerformIO m, Monad m)
                       => ([ParsedQuery] -> m a) -> FilePath -> m a
 withParsedQueriesFile p fn =
     withParsedFile
@@ -81,7 +81,7 @@ withParsedQueriesFile p fn =
         (p . nameQueries (queryIdentifier "" fn))
         fn
 
-withParsed :: (Monad m, Show e)
+withParsed :: (MonadFail m, Monad m, Show e)
            => (s -> Either e a) -> (a -> m b) -> s -> m b
 withParsed p a src = do
     let parseResult = p src
@@ -113,9 +113,8 @@ instance MonadPerformIO Q where
     addDependentFile = const $ return ()
 #endif
 
-withParsedFile :: (MonadPerformIO m, Monad m, Show e) => (String -> Either e a) -> (a -> m b) -> FilePath -> m b
+withParsedFile :: (MonadFail m, MonadPerformIO m, Monad m, Show e) => (String -> Either e a) -> (a -> m b) -> FilePath -> m b
 withParsedFile p a filename =
     addDependentFile filename >>
     performIO (readFile filename) >>=
         withParsed p a
-

--- a/hdbc/src/Database/YeshQL/HDBC/SqlRow/Class.hs
+++ b/hdbc/src/Database/YeshQL/HDBC/SqlRow/Class.hs
@@ -64,6 +64,8 @@ parserApply (Parser rpf) (Parser rpa) =
 
 instance Monad Parser where
     (>>=) = parserBind
+
+instance MonadFail Parser where
     fail = parserFail
 
 parserFail :: String -> Parser a
@@ -80,7 +82,7 @@ parserBind p f =
 class FromSqlRow a where
     parseSqlRow :: Parser a
 
-fromSqlRow :: (FromSqlRow a, Monad m) => [SqlValue] -> m a
+fromSqlRow :: (FromSqlRow a, Monad m, MonadFail m) => [SqlValue] -> m a
 fromSqlRow sqlValues =
     case runParser parseSqlRow sqlValues of
         Left err -> fail err

